msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -842,6 +842,8 @@ public static INativeAuthPublicClientApplication createNativeAuthPublicClientApp
                     null,
                     null
             );
+        }  catch (MsalException e) {
```

Comment:
MsalException is a type of BasException. The below clause was converting all MsalExceptions into unknown MsalException.